WalletWasabi/WabiSabi/Client/CredentialDependencies/RequestNode.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.WabiSabi.Client.CredentialDependencies
+{
+	// make private inner class of Graph?
```

Comment:
That is fine, less confusion in the big class. Usually, there is a folder in the project if there are more objects belong under one functionality.